File: Cpp/Algorithmic Solutions/MaximumSubArraySum/MaximumSubArraySum.cpp
#include <bits/stdc++.h>

using namespace std;

int maxSubArraySum(int*,int);

int main(){
    // int arr[10];
    string s;
    getline(cin,s);
    
    string delimiter = " ";
    size_t i = 0;
    while((i = s.find(delimiter))!=string::npos){
        cout<<s.substr(0,i)<<endl;
        s.erase(0,i+delimiter.length());
    }
    return 0;
}

int maxSubArraySum(int* arr,int len){
    int best = 0;
    int current = 0;
    for(int i=0; i<len; i++){
        current = max(arr[i],current+arr[i]);
        best = max(current, best);
    }
    return best;
}

File: Python/Algorithmic Solutions/MaximumSubArraySum/MaximumSubArraySum.py
import sys

def maxSubArray(arr):
    best = -sys.maxsize
    sum = 0
    
    for i in range(len(arr)):
        sum = max(arr[i],sum+arr[i])    
        best = max(best,sum)
    
    return best

if __name__ == "__main__":
    arr = [1,23,5,67,-34,65,-53,0]
    print(maxSubArray(arr))